src/client/sky.cpp

```diff
@@ -717,14 +703,108 @@ void Sky::draw_moon(video::IVideoDriver *driver, float moonsize, const video::SC
 	}
 }
 
+void Sky::draw_stars(video::IVideoDriver * driver, float wicked_time_of_day)
+{
+	driver->setMaterial(m_materials[1]);
+	// Tune values so that stars first appear just after the sun
+	// disappears over the horizon, and disappear just before the sun
+	// appears over the horizon.
+	// Also tune so that stars are at full brightness from time 20000
+	// to time 4000.
+
+	float tod = wicked_time_of_day < 0.5f ? wicked_time_of_day : (1.0f - wicked_time_of_day);
+	float starbrightness = clamp((0.25f - fabs(tod)) * 20.0f, 0.0f, 1.0f);
+
+	float f = starbrightness;
+	float d = (0.006 / 2) * m_star_params.scale;
+
+	video::SColor starcolor(f * m_star_params.starcolor.getAlpha(),
```

Comment:
Shorter:
```C++
video::SColor starcolor = m_star_params.starcolor;
starcolor.setAlpha(f * m_star_params.starcolor.getAlpha());

// Stars are only drawn when not fully transparent
if (starcolor < 1)
	return;
```

Reply:
I have the succinct feeling that `if (starcolor < 1)` would be invalid as the alpha channel is used to determine whether the stars aren't *fully invisible* due to alpha blending.